src/Stream.ts

```diff
@@ -31,31 +37,47 @@ export default class Stream<T> {
 
   constructor(
     client: Client,
-    topic: string,
+    topics: string[],
     messageTransformer: MessageTransformer<T>,
-    messageFilter?: MessageFilter
+    messageFilter?: MessageFilter,
+    contentTopicUpdater?: ContentTopicUpdater
   ) {
     this.messages = []
     this.resolvers = []
-    this.topic = topic
+    this.topics = topics
     this.client = client
-    this.callback = this.newMessageCallback(messageTransformer, messageFilter)
+    this.callback = this.newMessageCallback(
+      messageTransformer,
+      messageFilter,
+      contentTopicUpdater
+    )
   }
 
   // returns new closure to handle incoming Waku messages
   private newMessageCallback(
     transformer: MessageTransformer<T>,
-    filter?: MessageFilter
+    filter?: MessageFilter,
+    contentTopicUpdater?: ContentTopicUpdater
   ): (wakuMsg: WakuMessage) => Promise<void> {
     return async (wakuMsg: WakuMessage) => {
       if (!wakuMsg.payload) {
         return
       }
-      const msg = await this.client.decodeMessage(wakuMsg.payload)
+      const msg = await this.client.decodeMessage(
+        wakuMsg.payload,
+        wakuMsg.contentTopic
+      )
       // If there is a filter on the stream, and the filter returns false, ignore the message
       if (filter && !filter(msg)) {
         return
       }
+      // Check to see if we should update the stream's content topic subscription
+      if (contentTopicUpdater) {
+        const newTopics = contentTopicUpdater(msg)
+        if (newTopics) {
+          this.resubscribeToTopics(newTopics)
```

Comment:
`newTopics` is a bit confusing here. This is all the topics, right? The old ones and whatever new ones came in, correct?

Reply:
Oh ya, I was also a little thrown off by the variable name

Reply:
Great point! Yes, it's all topics including the new topic so I'll update the name for sure